### PR TITLE
Histogram as a stacked bar of individual points

### PR DESCRIPTION
**Pull request by JKinsman**
Addresses issue #1897 .  Assuming the amount of data visualized on the histogram is not unreasonable, it will behave as requested.

The feature will automatically disable if:
1.  Any series has more than fifty points in a single bin.  This metric was chosen because of limitations to the Highcharts API.  A highcharts series cannot have multiple data points with the same x-axis value.  Large bins from individual groups result in adding a considerable number of series to the chart, and quickly slow down performance.
2.  Any bin has more than one hundred points between all groups that are currently selected.  For similar reasons, bins with a large number of data points quickly degrade performance.  A threshold of 100 points was chosen because it seems to be around where older machines begin to slow down.

@jaypoulz

### CODE REVIEW

File: app/assets/javascripts/visualizations/highvis/histogram.coffee [hunk 1]
```diff
@@ -198,30 +198,63 @@ $ ->
...
+              # Get total for this bin
+              for dc, groupData of binObjs
+                if groupData[number]
+                  sum += groupData[number]
+
+              ret =
+                x: (Number number)
+                y: occurences
+                total: sum
+          ## --- ###
```
Comment (jaypoulz) [inline app/assets/javascripts/visualizations/highvis/histogram.coffee]: this comment can probably go :smile: I don't think It's been useful for a while haha.

File: app/assets/javascripts/visualizations/highvis/histogram.coffee [hunk 2]
```diff
@@ -198,30 +198,63 @@ $ ->
...
+                color: globals.configs.colors[Number(group) % globals.configs.colors.length],
+                name: data.groups[Number(group)]
+                data: binData
+              @chart.addSeries options, false
+              i += 1
+            i = 0
+        else
+          for groupIndex in data.groupSelection
+            finalData = for number, occurences of binObjs[groupIndex]
+
```
Comment (jaypoulz) [inline app/assets/javascripts/visualizations/highvis/histogram.coffee]: no newline here

**Comment by JKinsman** [comment]
@jaypoulz should be good to go once the newest commit passes Travis! :smile:

**Comment by jaypoulz** [comment]
I'm still opposed to this, but the patch is so nice  I can't help but r+.

One small note though - in the future, please keep lines < 80 characters.

**Pull request merged**